mlem/core/requirements.py

```diff
@@ -293,6 +293,26 @@ class UnixPackageRequirement(Requirement):
     package_name: str
 
 
+class CondaPackageRequirement(Requirement):
```

Comment:
they can be added only manually for now?

Reply:
Yes, How can we determine this automatically etc.? Could use some sense of direction in this.

Reply:
WDYM by manually? Also, why this is in `core`, while other parts are in `contrib`? Looks to me that `venv` is more "core" than `conda` TBH. Am I missing something?

Reply:
@aguschin: manually means our `get_requirements` method will never yield any `CondaPackageRequirement` and only way to have this in your metadata is to do smth like `model.requirements.add(...); model.update()`
And yes, it can be moved to `mlem.contrib.conda` and should work too (esp since there are no additional requirements for this extension).
@madhur-tandon: can give me an example of this conda requirements?
I can imagine 2 cases: 
- those are just python pip requirements, but you want to install them through conda. In this case you dont need a separate requirement implementation. Instead you would change CondaBuilder logic to try and install regular pip requirements with conda. Maybe you can check if those exist with something like `conda find` (not sure about it)
- There are some requirements that can be installed only through conda. In this case, can we determine them automatically? Probably not, since if it's known that eg `xgboost` package needs `libgomp1` package, it will have it as a dependency in conda ecosystem and we dont need to add this. I might be wrong here

Reply:
You can also add a field to CondaBuilder with list of additional CondaRequirements to install

Reply:
@mike0sv --
1. You shouldn't install packages that are meant to be installed via pip using conda. It is possible that some version is available on pip and not in conda. So we really shouldn't attempt to mix / just use conda instead of pip.

2. Yes, there are some reqs (usually non pythonic) which are only available through conda. I am not aware of how to determine those automatically. But of course, if we know beforehand -- we can add them explicitly, just like `UnixRequirements` are done in the `get_requirements` function currently.

Reply:
The decision is up to you guys, but it looks to me like we're adding some specific functionality with handling conda that is
1. Not that needed - if you want to spin up a new env for the model, venv should be enough. `conda` is something for Data Analysts / ML Developers - when you want to update your env from time to time, clean it, work with it for a long time, etc. When you just want to `serve` or `load` the model, it looks to me more like a temporary and separate env that will serve this purpose only. So `venv` is enough.
2.  Work when you do it manually and looks different from what we do with MLEM, by trying to do as many things automatically for the user as we could.

Consider this, I would suggest extracting conda functionality to a separate PR and leave it alone until we have a clear user request for this. Otherwise we're just wasting time and effort here, trying to come up with a decent solution for a user.

Reply:
Ok we can leave it as it is for now here - uses will have to add those manually. We can move conda req implementation to `contrib` and add an optional list of additional conda reqs for builder - that should suffice. Also create an issue to implement automagic conda detection for later

Reply:
ok so now I have moved it to `contrib` and it is now a list for `CondaBuilder` (with default as `[]` i.e. an empty list) aka it can be supplied by the user via command line I guess. This avoids playing around / modifying / or accessing the `requirements` attribute i.e. the user won't have to do `model_meta.requirements.add()`. So, with that, we don't have the other helper methods such as `self.of_type(CondaPackageRequirement)` or `_add_conda` since the user supplies a list directly to the builder. CC: @mike0sv @aguschin 

I hope this is the correct interpretation / conclusion of the discussion.
Otherwise I can revert the last commit.